hurain/Tic_tac_toe_del3.cpp

```diff
@@ -0,0 +1,185 @@
+                    if (cursorPos <= 5) cursorPos += 3;
+                    break;
+                case KEY_LEFT:
+                    if (cursorPos % 3 != 0) cursorPos -= 1;
+                    break;
+                case KEY_RIGHT:
+                    if (cursorPos % 3 != 2) cursorPos += 1;
+                    break;
+                case '\n':
+                case KEY_ENTER:
+                    if (board.spaces[cursorPos] == ' ') {
+                        board.spaces[cursorPos] = player;
+                        return;
+                    } else {
+                        printw("Spot taken, try another spot.\n");
+                        refresh();
+                        napms(1000);
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    void computerMove() {
+        int number;
+        srand(time(0));
+
+        while (true) {
+            number = rand() % 9;
+            if (board.spaces[number] == ' ') {
+                board.spaces[number] = computer;
+                break;
+            }
+        }
+    }
+
+    bool playAgain() {
+        char choice;
+        while (true) {
+            printw("Do you want to play again? (y/n): ");
+            refresh();
+            choice = getch();
+            if (choice == 'y' || choice == 'Y') {
+                return true;
+            } else if (choice == 'n' || choice == 'N') {
+                return false;
+            } else {
+                printw("Invalid input. Please enter 'y' or 'n'.\n");
+                refresh();
+                napms(1000);
+            }
+        }
+    }
+
+    void run() {
+        initscr();
+        keypad(stdscr, TRUE);
+        noecho();
+        curs_set(0);
+
+        bool running = true;
+        while (running) {
+            board = Board();
+            cursorPos = 0;
+            bool gameRunning = true;
+            board.drawBoard(cursorPos);
+
+            while (gameRunning) {
+                playerMove();
+                board.drawBoard(cursorPos);
+                if (board.checkWinner(player)) {
+                    printw("CONGRATULATIONS!!! YOU WIN!\n");
+                    gameRunning = false;
+                } else if (board.checkTie()) {
+                    gameRunning = false;
+                } else {
+                    computerMove();
+                    board.drawBoard(cursorPos);
+                    if (board.checkWinner(computer)) {
+                        printw("COMPUTER WINS! BETTER LUCK NEXT TIME!\n");
+                        gameRunning = false;
+                    } else if (board.checkTie()) {
+                        gameRunning = false;
+                    }
+                }
+            }
+            printw("Thanks for playing!\n");
+            running = playAgain();
+        }
+
+        endwin();
+    }
+};
+
+int main() {
+    Game game;
+    game.run();
+    return 0;
+}
```

Comment:
can you make the game del2 and del3 share the same base class or implementation? Use inheritance and polymorphi . The only difference is how the drawing of cursor and input.